web/src/user_profile.ts

```diff
@@ -1168,13 +1168,24 @@ export function show_edit_user_info_modal(user_id: number, $container: JQuery):
         const user_id = Number($("#edit-user-form").attr("data-user-id"));
         function handle_confirm(): void {
             const url = "/json/users/" + encodeURIComponent(user_id);
-            dialog_widget.submit_api_request(channel.del, url, {});
+            const is_spammer = $("#is-user-spammer").is(":checked");
+            const message_delete_action: Record<string, boolean> = {};
+            for (const action of settings_config.message_delete_action_values) {
+                message_delete_action[action.key] = $(
+                    `#message-delete-action[value='${action.code}']`,
+                ).is(":checked");
+            }
+            const data = {
+                is_spammer,
+                message_delete_action: JSON.stringify(message_delete_action),
+            };
+            dialog_widget.submit_api_request(channel.del, url, data);
```

Comment:
I think this should ideally be a function defined just once and not twice. Might be good to do it as a follow-up.